magenta/interfaces/midi/midi.py

```diff
@@ -498,48 +516,216 @@ def stop_capture(self):
     return self.captured_sequence
 
   @serialized
-  def wait_for_control_signal(self, control_number, control_value=None):
+  def wait_for_control_signal(self, control_message):
     """Blocks until a specific control signal arrives.
-
     Args:
-      control_number: The integer control change number.
-      control_value: The integer control change value or None if any is
-         acceptable.
-
+      control_message: The control change message.
     """
     if self._inport.callback is None:
-      # Not in a capture session.
+      # No callback set for inport
       for msg in self._inport:
-        if (msg.type == 'control_change' and msg.control == control_number and
-            (control_value is None or msg.value == control_value)):
+        if msg.hex()[:5] == control_message.hex()[:5] and msg.type == 'note_on':
+          if msg.velocity != 0:
+            msg = msg.copy(velocity=1)
+        if msg.hex() == control_message.hex():
           return
     else:
-      # In a capture session.
-      control_tuple = ((control_number,) if control_value is None else
-                       (control_number, control_value))
-      if control_tuple not in self._control_cvs:
-        self._control_cvs[control_tuple] = threading.Condition(self._lock)
-      self._control_cvs[control_tuple].wait()
-
-  def start_playback(self, sequence, metronome_velocity):
+      # In a capture session
+      if control_message.hex() not in self._control_cvs:
+        self._control_cvs[control_message.hex()] = threading.Condition(
+          self._lock)
+      self._control_cvs[control_message.hex()].wait()
+
+  def start_playback(self, sequence):
     """Plays the monophonic, sorted NoteSequence through the MIDI output port.
 
     Stops any previously playing sequences.
 
     Args:
       sequence: The monohponic, chronologically sorted NoteSequence to play.
-      metronome_velocity: The velocity of the metronome's MIDI note_on message.
     """
     self.stop_playback()
-    self._player = MonoMidiPlayer(self._outport, sequence, metronome_velocity)
+    self._player = MonoMidiPlayer(self._outport, sequence)
     self._player.start()
+    self._inport.callback = self._capture_message
 
   def stop_playback(self):
     """Stops any active sequence playback."""
     if self._player is not None and self._player.is_alive():
       self._player.stop()
       stdout_write_and_flush('Stopped\n')
 
+  def execute_cc_message(self, message):
+    """Defines how to treat non Start/Stop user defined CC messages.
+
+    Args:
+      message: The incoming Control Change message.
+    """
+
+    # Metronome Velocity
+    if message.bytes()[:2] == _CC_MAP['Metronome velocity'].bytes()[:2]:
+      global _METRONOME_VELOCITY
+      _METRONOME_VELOCITY = message.bytes()[2]
+
+class CCRemapper(object):
```

Comment:
Let's start the modularization of this file by moving this remapper to its own file.